tutorials/11-probabilistic-pca/11_probabilistic-pca.jmd

```diff
@@ -147,131 +164,79 @@ Note that:
 
 #### Step 3: Create the pPCA model
 
-```julia
-@model function pPCA(x, ::Type{TV}=Array{Float64}) where {TV}
-    # Dimensionality of the problem.
-    N, D = size(x)
-
-    # latent variable z
-    z ~ filldist(Normal(), D, N)
-
-    # side note for the curious
-    # we use the more concise filldist syntax partly for compactness, but also for compatibility with other AD
-    # backends, see the [Turing Performance Tipps](https://turing.ml/dev/docs/using-turing/performancetips)
-    # w = TV{2}(undef, D, D)
-    # for d in 1:D
-    #  w[d, :] ~ MvNormal(ones(D))
-    # end
-
-    # weights/loadings W
-    w ~ filldist(Normal(), D, D)
-
-    # mean offset
-    m ~ MvNormal(ones(D))
-    mu = (w * z .+ m)'
-    for d in 1:D
-        x[:, d] ~ MvNormal(mu[:, d], ones(N))
-    end
-end;
-```
+Here we construct the probabilistic model `pPCA()`. As per the p-PCA formula, we think of each row (i.e. each gene feature) following a $N=60$ dimensional multivariate normal distribution centered around the corresponding row of $\mathbf{W}_{D \times k} \times \mathbf{Z}_{k \times N} + \boldsymbol{\mu}_{D \times N}$.
 
 ```julia
-@model function pPCA(x, k)
-    # function pPCA(k::Int64, x::Type{TV}=Array{Float64}) where {TV}
+@model function pPCA(X::Array{Float64}, k::Int64)
     # retrieve the dimension of input matrix x.
-    N, D = size(x)
+    # N, D = size(x)
+    D, N = size(X)
```

Comment:
Again it seems the dimensions are flipped? Why? In any case, the comment is confusing and should be removed:
```suggestion
    D, N = size(X)
```

Reply:
yes, the usages of transpose in the old and this version are confusing. After a thorough review, I use the following rule:
1. In data generation, always use features as rows, instances as columns (i.e. `X` is of dimension `D by N` where `D` is the number of feature, `N` is the number of instances/examples/observations) - this is consistent with your suggestion, accessing columns is accessing the examples. However, in some textbooks (e.g. Stephen Boyds' <Linear Algebra and Least Squares>) and the wide data science community (e.g. Kaggle), people tend to define the `design matrix` X as instances by row and features by column, which is also commonly used in stats community (e.g. the Gram matrix (X^TX) in least squares linear regression).
2. in model specification `pPCA` and `pPCA_ARD`, the input argument `X` is of dimension `N by D`, which is a transpose of the data matrix. This is mainly to comply with the ```X ~ arraydist([MvNormal(m, Eye(N)) for m in eachcol(genes_mean')])``` setting, because accessing each column is quicker than accessing each row - as you suggested.